.github/workflows/autotest.yml

```diff
@@ -1,12 +1,14 @@
 name: Run all the autotests
 
 on:
-  - pull_request
-  - push
+  pull_request:
+  push:
+  schedule:
+    - cron: '0 3 * * *'
 
 jobs:
   header-lint:
-    if: 'github.event_name == ''pull_request'''
+    if: ${{ 'github.event_name == ''pull_request''' || 'github.event_name == ''schedule''' }}
```

Comment:
We create a diff between the current branch and the master branch to get the changed files. In the case of a scheduled run this would not be possible, since the master branch is the current branch.

I would only run this job on every pull request.